semgrep-core/src/parsing/tree_sitter/Parse_ocaml_tree_sitter.ml

```diff
@@ -104,1272 +89,1260 @@ let map_anon_choice_EQ_4ccabd6 (env : env) (x : CST.anon_choice_EQ_4ccabd6) =
-  | `Pat_60fc52b tok -> token env tok (* pattern "\\\\[\\\\\"'ntbr ]" *)
-  | `Pat_86b875b tok -> token env tok (* pattern \\[0-9][0-9][0-9] *)
-  | `Pat_3bf11d1 tok -> token env tok (* pattern \\x[0-9A-Fa-f][0-9A-Fa-f] *)
-  | `Pat_21333c0 tok -> token env tok
+  | `Pat_60fc52b tok -> str env tok (* pattern "\\\\[\\\\\"'ntbr ]" *)
+  | `Pat_86b875b tok -> str env tok (* pattern \\[0-9][0-9][0-9] *)
+  | `Pat_3bf11d1 tok -> str env tok (* pattern \\x[0-9A-Fa-f][0-9A-Fa-f] *)
+  | `Pat_21333c0 tok -> str env tok
 
 (* pattern \\o[0-3][0-7][0-7] *)
 
 let map_pow_operator (env : env) (x : CST.pow_operator) =
   match x with
   | `Tok_STARSTAR_rep_pat_525fbb4 tok ->
-      token env tok (* tok_STARSTAR_rep_pat_525fbb4 *)
-  | `Lsl tok -> token env tok (* "lsl" *)
-  | `Lsr tok -> token env tok (* "lsr" *)
-  | `Asr tok -> token env tok
+      str env tok (* tok_STARSTAR_rep_pat_525fbb4 *)
+  | `Lsl tok -> str env tok (* "lsl" *)
+  | `Lsr tok -> str env tok (* "lsr" *)
+  | `Asr tok -> str env tok
 
 (* "asr" *)
 
 let map_quoted_string_content (env : env) (xs : CST.quoted_string_content) =
   List.map
     (fun x ->
       match x with
-      | `Imm_tok_SPACE tok -> token env tok (* " " *)
-      | `Imm_tok_LF tok -> token env tok (* "\n" *)
-      | `Imm_tok_HT tok -> token env tok (* "\t" *)
-      | `Imm_tok_LBRACKAT tok -> token env tok (* "[@" *)
-      | `Imm_tok_LBRACKATAT tok -> token env tok (* "[@@" *)
-      | `Imm_tok_LBRACKATATAT tok -> token env tok (* "[@@@" *)
-      | `Pat_714c625 tok -> token env tok (* pattern [^%@|]+|%|@|\| *)
-      | `Null tok -> token env tok (* null *)
-      | `Conv_spec tok -> token env tok (* conversion_specification *)
-      | `Pretty_prin_indi tok -> token env tok
+      | `Imm_tok_SPACE tok -> str env tok (* " " *)
+      | `Imm_tok_LF tok -> str env tok (* "\n" *)
+      | `Imm_tok_HT tok -> str env tok (* "\t" *)
+      | `Imm_tok_LBRACKAT tok -> str env tok (* "[@" *)
+      | `Imm_tok_LBRACKATAT tok -> str env tok (* "[@@" *)
+      | `Imm_tok_LBRACKATATAT tok -> str env tok (* "[@@@" *)
+      | `Pat_714c625 tok -> str env tok (* pattern [^%@|]+|%|@|\| *)
+      | `Null tok -> str env tok (* null *)
+      | `Conv_spec tok -> str env tok (* conversion_specification *)
+      | `Pretty_prin_indi tok -> str env tok
       (* pattern @([\[\], ;.{}?]|\\n|<[0-9]+>) *))
     xs
 
-let map_constructor_name (env : env) (x : CST.constructor_name) =
+let map_constructor_name (env : env) (x : CST.constructor_name) : ident =
   match x with
-  | `Capi_id tok -> token env tok (* pattern "[A-Z][a-zA-Z0-9_']*" *)
+  | `Capi_id tok -> str env tok (* pattern "[A-Z][a-zA-Z0-9_']*" *)
   | `LPAR_COLONCOLON_RPAR (v1, v2, v3) ->
-      let v1 = token env v1 (* "(" *) in
-      let v2 = token env v2 (* "::" *) in
-      let v3 = token env v3 (* ")" *) in
-      todo env (v1, v2, v3)
+      let _v1 = token env v1 (* "(" *) in
+      let v2 = str env v2 (* "::" *) in
+      let _v3 = token env v3 (* ")" *) in
+      v2
```

Comment:
It would be great if you could give these variables a name while it's fresh in your mind e.g. `let coloncolon = str env v2 in ... coloncolon`. Having to figure out their meaning later is tedious.